script.service.playbackresumer/default.py

```diff
@@ -0,0 +1,449 @@
+    log("VideoLibrary." + method + " response: " + json.dumps(jsonResponse))
+
+
+# Persistently tracks the currently playing file (in case of crash, for possible resuming)
+
+def updateCurrentPlayingFilePath(filepath):
+
+    global currentPlayingFilePath
+    global libraryId
+    global typeOfVideo
+    
+    if isExcluded(filepath):
+        log("Skipping excluded filepath: " + filepath)
+        currentPlayingFilePath = ''
+        return
+    
+    currentPlayingFilePath = filepath
+    
+    # write the full path to a file for persistent tracking
+    f = open(lastPlayedTrackerFilePath, 'w+')
+    f.write(filepath)
+    f.close()
+    log('Last played file set to: "%s"' % filepath)
+    
+    # check if its a library video and get the libraryId and typeOfVideo
+    
+    query = {
+        "jsonrpc": "2.0",
+        "method": "Files.GetFileDetails",
+        "params": {
+            "file": filepath,
+            "media": "video",
+            "properties": [
+                "playcount",
+                "runtime"
+            ]
+        },
+        "id": "fileDetailsCheck"
+    }
+    
+    log("Executing JSON-RPC: " + json.dumps(query))
+    jsonResponse = json.loads(xbmc.executeJSONRPC(json.dumps(query)))
+    log("Files.GetFileDetails response: " + json.dumps(jsonResponse))
+    
+    typeOfVideo = 'unknown'
+
+    try:
+        typeOfVideo = jsonResponse['result']['filedetails']['type']
+    except:
+        libraryId = -1
+        log("Could not determine type of video; assuming video is not in Kodi's library: " + currentPlayingFilePath)
+
+    if typeOfVideo == 'episode' or typeOfVideo == 'movie' or typeOfVideo == 'musicvideo':
+        libraryId = jsonResponse['result']['filedetails']['id']
+        log("The libraryid for this " + typeOfVideo + " is " + str(libraryId))
+    else:
+        libraryId = -1
+        log("This type of video is not supported for resume points because it is not in Kodi's library: " + typeOfVideo + ": " + currentPlayingFilePath)
+
+
+# Automatically resume a video after a crash, if one was playing...
+def resumeIfWasPlaying():
+
+    global resumeonstartup
+    global xbmc_monitor
+
+    if resumeonstartup and os.path.exists(resumePointTrackerFilePath) and os.path.exists(lastPlayedTrackerFilePath):
+        
+        f = open(resumePointTrackerFilePath, 'r')
+        resumePoint = float(f.read())
+        f.close()
+        
+        # neg 1 means the video wasn't playing when Kodi ended
+        if resumePoint < 0:
+            log("Not resuming playback because nothing was playing when Kodi ended")
+            return False
+
+        f = open(lastPlayedTrackerFilePath, 'r')
+        fullPath = f.read()
+        f.close()
+
+        strTimestamp = str(int(resumePoint / 60))+":"+("%02d" % (resumePoint % 60))
+        
+        log("Will resume playback at "+ strTimestamp+" of " + fullPath)
+        
+        xbmc.Player().play(fullPath)
```

Comment:
Better to create a player instance and reuse it instead of creating a new one all the time using xbmc.Player(). This is an expensive operation